contracts/AddressBook.sol

```diff
@@ -3,10 +3,119 @@
  */
 pragma solidity 0.6.10;
 
+import {Ownable} from "./packages/oz/Ownable.sol";
+import {OwnedUpgradeabilityProxy} from "./packages/openzeppelin-upgradeability/OwnedUpgradeabilityProxy.sol";
+
 /**
- *
+ * @author Opyn Team
+ * @title AddressBook Module
  */
-// solhint-disable-next-line no-empty-blocks
-contract AddressBook {
+contract AddressBook is Ownable {
+    bytes32 private constant OTOKEN_IMPL = "OTOKEN_IMPL";
+    bytes32 private constant OTOKEN_FACTORY = "OTOKEN_FACTORY";
+    bytes32 private constant MARGIN_WHITELIST = "MARGIN_WHITELIST";
+    bytes32 private constant CONTROLLER = "CONTROLLER";
+    bytes32 private constant MARGIN_POOL = "MARGIN_POOL";
+    bytes32 private constant MARGIN_CALCULATOR = "MARGIN_CALCULATOR";
+    bytes32 private constant LIQUIDATION_MANAGER = "LIQUIDATION_MANAGER";
+    bytes32 private constant ORACLE = "Oracle";
+
+    /// @dev a mapping between module key and module address
+    mapping(bytes32 => address) private modules;
+
+    /// @notice event emitted when a new proxy get created
+    event ProxyCreated(bytes32 id, address proxy);
+
+    function getAddress(bytes32 _key) public view returns (address) {
+        return modules[_key];
+    }
+
+    function getOtokenImpl() external view returns (address) {
+        return getAddress(OTOKEN_IMPL);
+    }
+
+    function getOtokenFactory() external view returns (address) {
+        return getAddress(OTOKEN_FACTORY);
+    }
+
+    function getWhitelist() external view returns (address) {
+        return getAddress(MARGIN_WHITELIST);
+    }
+
+    function getController() external view returns (address) {
+        return getAddress(CONTROLLER);
+    }
+
+    function getMarginPool() external view returns (address) {
+        return getAddress(MARGIN_POOL);
+    }
+
+    function getMarginCalculator() external view returns (address) {
+        return getAddress(MARGIN_CALCULATOR);
+    }
+
+    function getLiquidationManager() external view returns (address) {
+        return getAddress(LIQUIDATION_MANAGER);
+    }
+
+    function getOracle() external view returns (address) {
+        return getAddress(ORACLE);
+    }
+
+    function setAddress(bytes32 _key, address _module) public onlyOwner {
+        modules[_key] = _module;
+    }
+
+    function setOtokenImpl(address _otokenImpl) external onlyOwner {
+        updateImplInternal(OTOKEN_IMPL, _otokenImpl);
```

Comment:
We don't need proxy for `otokenImpl`. The `Otoken.sol` contract is not upgradable.